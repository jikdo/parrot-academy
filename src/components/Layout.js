import * as React from 'react';
import { graphql, Link, StaticQuery } from 'gatsby';
// import "@fontsource/roboto";
import { container, navHeading } from './layout.module.css';

const Layout = ({ data , pageTitle, children }) => {

    return (
        <div className={container}>
            <title>{data.site.siteMetadata.title} | {pageTitle}</title>
            <nav>
                <p className={navHeading}><Link to="/">Parrot Academy</Link></p>
            </nav>

            <main>
                {children}
            </main>

        </div>
    )
}

export default function WrapperLayout(props) {
    return(
        <StaticQuery
        query={graphql`
        query {
            site {
                siteMetadata {
                    title
                }
            }
        }
        `}
        render={data => <Layout data={data} {...props} /> }
        />
    )
}