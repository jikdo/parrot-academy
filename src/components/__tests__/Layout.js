import React from "react";
import renderer from 'react-test-renderer';
import { StaticQuery } from "gatsby";
import Layout from "../Layout";

beforeEach(() => {
    StaticQuery.mockImplementationOnce(({ render }) =>
        render({
            site: {
                siteMetadata: {
                    title: `Parrot Academy`,
                },
            },
        })
    )
})


describe("Layout", () =>
    it("renders correctly", () => {
        const tree = renderer.create(<Layout pageTitle=""></Layout>).toJSON()
        expect(tree).toMatchSnapshot()
    }))