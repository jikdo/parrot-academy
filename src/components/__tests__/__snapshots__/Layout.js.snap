// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="container"
>
  <title>
    Parrot Academy
     | 
    
  </title>
  <nav>
    <p
      className="navHeading"
    >
      <a
        href="/"
      >
        Parrot Academy
      </a>
    </p>
  </nav>
  <main />
</div>
`;
